be/src/exprs/vectorized/time_functions.cpp

```diff
@@ -1253,9 +1254,11 @@ ColumnPtr date_format_func(const Columns& cols, size_t patten_size) {
     ColumnBuilder<TYPE_VARCHAR> builder;
     ColumnViewer<Type> viewer(cols[0]);
 
-    builder.data_column()->reserve(viewer.size(), viewer.size() * patten_size);
+    size_t num_rows = viewer.size();
```

Comment:
If the viewer.size() is not right when column is const.
Why not remove `size` method from ColumnViewer?

Reply:
For unary functions, viewer.size() is very useful. for const column, we only need to calculate once and return a const column.
This simplifies a lot of the logic code. such as some case function. Otherwise, if we want to achieve such an optimization we need to add the following additional code.

```
if (column[0]->isconstant()) {
   // do some cast 
  return buildConst
} else {
  // do some cast
  return result_column
}
```

So there is further discussion on whether to remove ColumnViewer::size